lemonspotter/core/argument.py

```diff
@@ -0,0 +1,46 @@
+"""
+This module contains the class definition of an Argument.
+"""
+
+from typing import Sequence, Optional
+
+from lemonspotter.core.variable import Variable
+
+
+class Argument:
+    """
+    This class represents parameters of Function objects translated from the specification.
+    """
+
+    # TODO Arguments need to also support code fragements?
+    # some dependencies will require executing additional code.
+
+    def __init__(self,
+                 variable: Variable,
+                 variables: Optional[Sequence[Variable]] = None):
+        self._variable = variable
+        self._variables = variables
```

Comment:
What is the difference between variable and variables?

Reply:
Will rename to _dependencies.